src/utils/size-observer.js

```diff
@@ -114,19 +174,49 @@ function processEntries(entries) {
     if (!callbacks) {
       continue;
     }
-    const {width, height} = contentRect;
-    /** @type {../layout-rect.LayoutSizeDef} */
-    const size = {width, height};
-    targetSizeMap.set(target, size);
+    targetEntryMap.set(target, entry);
     for (let k = 0; k < callbacks.length; k++) {
-      callCallbackNoInline(callbacks[k], size);
+      const {type, callback} = callbacks[k];
+      computeAndCall(type, callback, entry);
     }
   }
 }
 
+/**
+ * @param {Type} type
+ * @param {!ObserverCallbackDef} callback
+ * @param {!ResizeObserverEntry} entry
+ */
+function computeAndCall(type, callback, entry) {
+  let width, height;
+  if (type == Type.CONTENT) {
+    const {contentRect} = entry;
+    width = contentRect.width;
+    height = contentRect.height;
+  } else if (type == Type.BORDER) {
+    const {borderBoxSize} = entry;
+    if (borderBoxSize?.length > 0) {
+      // `borderBoxSize` is supported. Only single-fragment border boxes are
+      // supported here (`borderBoxSize[0]`).
+      width = borderBoxSize[0].inlineSize;
```

Comment:
Does `inlineSize` always map to `width`?

Reply:
In classical yes. Basically in the normal writing mode `inlineSize` is the width. The "inline" term is confusing. What it really means is this: as you print uninterrupted characters into, say, a `div`, eventually they'd hit the boundary and start wrapping. That boundary is, in a classical world, a `width`.

Reply:
The reason for the distinction: for a vertical language, the roles become reversed (inline would map to height).

In order for this to be correct, we also need to check the [writing-mode](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode) of the observed element

Reply:
I'd say "no". Just as `offsetWidth` and `offsetHeight` names don't change, so our `width`/`height` don't need to change either. That terminology clarifies different writing modes better, but if we deal with `width`/`height` naming we're still ok. It's just a bit confusing - that's all. Not unlike `left` and `start` terms in CSS.

Reply:
Alternatively to unconfuse this we can rely on the fact that we only need `width x height` in a way. I.e. we actually don't care for pausing which dimension is which as long as we know they are both non-zero.

Reply:
To clarify: what we'd do in this case - we'd just propagate `ResizeObserverEntry` from the `size-observer` API to `PauseHelper`, which will interpret the `borderBox` as it sees fit. That way we don't need any generic logic for it, until we're ready.

Reply:
I'm probably misunderstanding. LMK if I have this right:

For horizontal `writing-mode`:
- inlineSize --> offsetWidth
- blockSize --> offsetHeight

For a vertical `writing-mode`:
- inlineSize --> offsetHeight
- blockSize --> offsetWidth

Therefore in order for our height/width we gather to be correct, we need to correctly assign them

Reply:
> Just as offsetWidth and offsetHeight names don't change, so our width/height don't need to change either.

I think Jake's point is the polyfill case, we're correctly creating `{ width: offsetWidth, height: offseHeight }`. But in native case, we don't know whether `inlineSize` corresponds with `width` or `height`. We'd need to either fix polyfill to translate `width`/`height` to `inlineSize`/`blocksize`, or fix native case to translate the other way. That way.

Or, we could create an "area", multiply the two numbers together, and forget width/height exist.

Reply:
My suggestion is to remove this interpretation from this class for now and expose this data simply to the `PauseHelper` which will interpret it as `dimension1 > 0 && dimension2 > 0` w/o worrying about giving them the correct names and possibly propagating confusion downstream.

Reply:
Sure, changing the name to not imply `width`/`height`/`inlineSize`/`blockSize` works for me.

Reply:
Here's what it'd look like: https://github.com/ampproject/amphtml/pull/33622. LMK what you think is better.

Reply:
And, of course, measuring writing mode would significantly reduce the benefits of non-blocking `ResizeObserver`. LMK what you think.

Reply:
I think that's definitely worse, because we're passing the code to the callers. What I'm looking for is either change callback param's struct to ` { dimension1: number, dimension1: number }` so that width/height are not implied, or change, or we can change the polyfill to measure `inlineSize`/`blockSize` and pass ` { inlineSize: number, blockSize: number }` to callers of `observeBoxSize`.

Reply:
But this way we simply pass the `ResizeObserverEntry` structure as is and letting the caller to decide how to interpret that. That means the caller can make a better decision. No?

Reply:
I think it complicates the caller, because it now has to handle the `borderBoxSize` isn't implemented case. It happens to work in this case because the caller doesn't really care about any of the sizes, only that the sizes are non-zero.

Reply:
Done. PTAL.